### PR TITLE
Add cert generation lib and tool.

### PR DESCRIPTION
**Pull request by wlu2016**
@lookuptable

### CODE REVIEW

File: cert_manager/generate_cert.go [hunk 1]
```diff
@@ -0,0 +1,188 @@
...
+)
+
+type CertOptions struct {
+	// Comma-separated hostnames and IPs to generate a certificate for.
+	// This can also be set to the identity running the workload,
+	// like kubernetes service account.
+	Host string
+
+	// Creation date formatted as Jan 1 15:04:05 2011.
+	ValidFrom string
```
Comment (lookuptable) [inline cert_manager/generate_cert.go]: Why 'ValidFrom` is a string but `ValidFor` is a time.Duration?

File: cert_manager/generate_cert.go [hunk 2]
```diff
@@ -0,0 +1,188 @@
...
+	// case, otherwise the certificate is signed by the signer private key
+	// as specified in the CertOptions.
+	priv, err := rsa.GenerateKey(rand.Reader, RsaBits)
+	if err != nil {
+		log.Fatalf("RSA key generation failed with error %s.", err)
+	}
+	template := genCertTemplate(options)
+	signingCert, signingKey := &template, priv
+	if !options.IsSelfSigned {
+		signingCert, signingKey = loadSigningCreds(options.SignerPriv, options.SignerCert)
```
Comment (lookuptable) [inline cert_manager/generate_cert.go]: It feels strange that the signing key and cert are passed in by the caller. Instead I think the CA should generate/read from a file during startup and use the same key & cert to sign all requests.

File: cert_manager/generate_cert.go [hunk 3]
```diff
@@ -0,0 +1,188 @@
...
+
+	// Whether this cerificate is self-signed.
+	IsSelfSigned bool
+
+	// Whether this certificate is for a client.
+	IsClient bool
+}
+
+// Size of RSA key to generate.
+const RsaBits = 2048
```
Comment (lookuptable) [inline cert_manager/generate_cert.go]: Should this just be private?

File: cert_manager/generate_cert.go (hunk 1 above)
Comment (wlu2016) [inline cert_manager/generate_cert.go]: ValidFrom can be empty, default to time.Now()

**Pull request merged**